### PR TITLE
Mobile dumb proofs DESKTOP-1171

### PR DESCRIPTION
**Pull request by awendland**
This mostly takes advantage of the desktop proofs code, and simply modifies it to be mobile appropriate. Shared code is extracted and placed in `user-proofs.shared.js`. Some cleanup was done around the mobile tracker and user-bio so that this component could be tested. The cleanup is minimal and expected to be changed by proper implementations of those components.

An attempt was made at shimming padding on Text elements so that it worked on Android as well as iOS, but this resulted in unideal looking visual output because there are other inconsistencies in how view sizes/margins are calculated. This was reverted, and a wrapper was manually created around the Text view instead. This wrapper has special values applied for iOS vs. Android because of discrepancies that arose.

External linking should work. Because the links are based on the Android intent system (and the iOS equivalent) the URLs sometimes have to be modified to contain protocols if they are not otherwise present so that the OS knows to use a browser to view them.

### Example valid proofs on both mobile OSs:

iOS:
screen shot 2016-06-22 at 4 07 09 pm
Android:
screen shot 2016-06-22 at 4 07 16 pm

### Example meta tags on Android (looks the same on iOS)
screen shot 2016-06-23 at 12 52 21 pm

### Example you track them on Android (looks the same on iOS)
screen shot 2016-06-22 at 4 25 21 pm

@keybase/react-hackers

### CODE REVIEW

File: shared/common-adapters/tab-bar.native.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class TabBarButton extends Component<void, TabBarButtonProps, void> {
     return (
       <Box style={{...globalStyles.flexBoxColumn, backgroundColor, ...stylesTabBarButtonIcon, ...this.props.style}}>
         {this.props.source.type === 'icon'
-          ? <Icon type={this.props.source.icon} style={{fontSize: 48, fontWidth: 80, textAlign: 'center', color: this.props.selected ? globalColors.blue3 : globalColors.blue3_40, ...this.props.styleIcon}} />
+          ? <Icon type={this.props.source.icon} style={{fontSize: 48, width: 80, textAlign: 'center', color: this.props.selected ? globalColors.blue3 : globalColors.blue3_40, ...this.props.styleIcon}} />
```
Comment (awendland) [inline shared/common-adapters/tab-bar.native.js]: I removed fontWidth as a prop because it was not being used properly depending on the icon type, and was overwriting the value provided in styles. It also doesn't exist on desktop.

File: shared/common-adapters/text.native.js [hunk 2]
```diff
@@ -132,13 +132,11 @@ export default class Text extends Component {
     const style = Text.textStyle(this.props, this.context)
 
     const terminalPrefix = this._terminalPrefix(this.props.type)
-    const className = this.props.className
```
Comment (awendland) [inline shared/common-adapters/text.native.js]: This property is never utilized in react-native

File: shared/common-adapters/user-proofs.desktop.js [hunk 3]
```diff
@@ -1,110 +1,35 @@
...
+import * as shared from './user-proofs.shared'
+import {metaNone} from '../constants/tracker'
+import {checking as proofChecking} from '../constants/tracker'
 
 import type {Proof, Props} from './user-proofs'
 
 class ProofsRender extends Component {
   props: Props;
 
-  _openLink (url: string): void {
```
Comment (awendland) [inline shared/common-adapters/user-proofs.desktop.js]: Using the util function instead

File: shared/common-adapters/user-proofs.shared.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    default:
+      return null
+  }
+}
+
+export function proofNameStyle (proof: Proof) {
+  return {
+    color: proofColor(proof),
+    ...(proof.meta === metaDeleted ? {textDecorationLine: 'line-through'} : {}),
```
Comment (awendland) [inline shared/common-adapters/user-proofs.shared.js]: This property is only supported on iOS

File: shared/styles/style-guide.native.js [hunk 5]
```diff
@@ -92,3 +92,12 @@ export const globalStyles = {
...
 }
+
+export const globalMargins = {
+  xtiny: 4,
+  tiny: 8,
+  small: 16,
+  medium: 32,
+  large: 48,
+  xlarge: 64,
+}
```
Comment (awendland) [inline shared/styles/style-guide.native.js]: To match the desktop style guide

File: shared/tracker/dumb.native.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    'NonuserLink': trackerPropsToRenderProps(propsNonUser),
+    'NonuserNoLinkPublic': trackerPropsToRenderProps({...propsNonUser, inviteLink: null}),
+    'Logged out': trackerPropsToRenderProps(propsLoggedOut),
+  },
+}
+
+export default {
+  'Tracker': dumbMap,
+}
+
```
Comment (awendland) [inline shared/tracker/dumb.native.js]: All stolen from the desktop dumb component. Maybe these should be combined? Though this one doesn't use parent props ever.

File: shared/tracker/render.desktop.js [hunk 7]
```diff
@@ -50,13 +50,6 @@ export default class Render extends Component<void, RenderProps, void> {
   }
 }
 
-Render.propTypes = {
-  headerProps: React.PropTypes.any,
-  bioProps: React.PropTypes.any,
-  proofsProps: React.PropTypes.any,
-  actionProps: React.PropTypes.any,
-}
-
```
Comment (awendland) [inline shared/tracker/render.desktop.js]: Outdated now that we use Flow

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits f8bd1e628d2c...810258550247 introduced some visual changes:
 * Changed: **Tracker-Changed_Broken_proofs_user_you_dont_follow** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Tracker-Changed_Broken_proofs_user_you_dont_follow-f8bd1e628d2c.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Tracker-Changed_Broken_proofs_user_you_dont_follow-810258550247.png)  
   Tracker-Changed_Broken_proofs_user_you_dont_follow comparison
 * Changed: **Tracker-Changed_Broken_proofs** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Tracker-Changed_Broken_proofs-f8bd1e628d2c.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Tracker-Changed_Broken_proofs-810258550247.png)  
   Tracker-Changed_Broken_proofs comparison
 * Changed: **Devices_List-Devices** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Devices_List-Devices-f8bd1e628d2c.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466638982424-f8bd1e628d2c-810258550247/Devices_List-Devices-810258550247.png)  
   Devices_List-Devices comparison

File: shared/common-adapters/user-proofs.native.js [hunk 8]
```diff
@@ -1,32 +1,115 @@
...
+    paddingBottom: 16,
+    alignItems: 'stretch',
+    paddingLeft: globalMargins.medium,
+    paddingRight: globalMargins.medium,
+  },
+  row: {
+    ...globalStyles.flexBoxRow,
+    alignItems: 'flex-start',
+    justifyContent: 'flex-start',
+    // RN-BUG: set maxWidth once that prop is supported
```
Comment (awendland) [inline shared/common-adapters/user-proofs.native.js]: I'm using the tag `RN-BUG` for easy lookup of react-native hacks/todos/inconsistencies in the code base

File: shared/common-adapters/user-proofs.shared.js (hunk 4 above)
Comment (chrisnojima) [inline shared/common-adapters/user-proofs.shared.js]: I think this is supported on android now

Comment (chrisnojima) [inline shared/common-adapters/user-proofs.shared.js]: yeah it is https://facebook.github.io/react-native/docs/text.html#style

Comment (awendland) [inline shared/common-adapters/user-proofs.shared.js]: I'll need to test further to see why it's only showing up on iOS not Android then. Maybe I'm not working off of master with the latest RN.

Comment (chrisnojima) [inline shared/common-adapters/user-proofs.shared.js]: yeah you have to be updated, our rn version previously didn't have it

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits a6bd5d62e55b...e57605f58bb8 introduced some visual changes:
 * Changed: **Tracker-Changed_Broken_proofs_user_you_dont_follow** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466711538805-a6bd5d62e55b-e57605f58bb8/Tracker-Changed_Broken_proofs_user_you_dont_follow-a6bd5d62e55b.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466711538805-a6bd5d62e55b-e57605f58bb8/Tracker-Changed_Broken_proofs_user_you_dont_follow-e57605f58bb8.png)  
   Tracker-Changed_Broken_proofs_user_you_dont_follow comparison
 * Changed: **Tracker-Changed_Broken_proofs** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466711538805-a6bd5d62e55b-e57605f58bb8/Tracker-Changed_Broken_proofs-a6bd5d62e55b.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466711538805-a6bd5d62e55b-e57605f58bb8/Tracker-Changed_Broken_proofs-e57605f58bb8.png)  
   Tracker-Changed_Broken_proofs comparison

File: shared/common-adapters/icon.native.js [hunk 9]
```diff
@@ -26,19 +26,17 @@ export default class Icon extends Component {
...
     const textAlign = this.props.style && this.props.style.textAlign
-    const fontWidth = this.props.style && this.props.style.fontWidth
 
     // Color is for our fontIcon and not the container
     let containerProps = {...this.props.style}
     delete containerProps.color
     delete containerProps.width
     delete containerProps.height
     delete containerProps.textAlign
-    delete containerProps.fontWidth
```
Comment (MarcoPolo) [inline shared/common-adapters/icon.native.js]: nvm saw your note

File: shared/common-adapters/user-proofs.native.js [hunk 10]
```diff
@@ -1,32 +1,115 @@
...
+import {Icon, Meta} from '../common-adapters/index'
+import {globalStyles, globalColors, globalMargins} from '../styles/style-guide'
+import {checking as proofChecking} from '../constants/tracker'
+
+import type {Props, Proof} from './user-proofs'
+
 export default class ProofsRender extends Component {
+  props: Props;
+
+  _ensureUrlProtocal (url: string): string {
```
Comment (MarcoPolo) [inline shared/common-adapters/user-proofs.native.js]: Why is this needed? Desktop doesn't have this, right?

File: shared/common-adapters/user-proofs.native.js [hunk 11]
```diff
@@ -1,32 +1,115 @@
...
-        <Text>Proof20</Text>
-        <Text>Proof21</Text>
+      <View style={styles.container}>
+        {this.props.proofs.map((p, idx) => this._renderProofRow(p, idx))}
       </View>
     )
   }
 }
+
+const styles = {
```
Comment (MarcoPolo) [inline shared/common-adapters/user-proofs.native.js]: prefer doing `stylesContainer = { ... }` and `stylesRow = { ... }`

File: shared/tracker/dumb.native.js (hunk 6 above)
Comment (MarcoPolo) [inline shared/tracker/dumb.native.js]: If parentProps is the only difference maybe they should be combined and parentProps would be behind a platform check

File: shared/common-adapters/user-proofs.native.js (hunk 10 above)
Comment (awendland) [inline shared/common-adapters/user-proofs.native.js]: Desktop doesn't have this because electron will always default to using a browser to open the external links. Native tries to accommodate the links depending on their content, and so needs the hint of the http(s) to indicate that the browser should handle it.

File: shared/common-adapters/user-proofs.native.js (hunk 11 above)
Comment (awendland) [inline shared/common-adapters/user-proofs.native.js]: Reason for changing to the way mentioned by Marco: ESLint will pick up if there are unused styles if the styles are top level variables instead of nested inside an object.

**Comment by MarcoPolo** [comment]
👍

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 545595487b2d...9a33b32608e5 introduced some visual changes:
 * Changed: **Tracker-Changed_Broken_proofs_user_you_dont_follow** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466715971459-545595487b2d-9a33b32608e5/Tracker-Changed_Broken_proofs_user_you_dont_follow-545595487b2d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466715971459-545595487b2d-9a33b32608e5/Tracker-Changed_Broken_proofs_user_you_dont_follow-9a33b32608e5.png)  
   Tracker-Changed_Broken_proofs_user_you_dont_follow comparison
 * Changed: **Tracker-Changed_Broken_proofs** [(before)](http://keybase-app-visdiff.s3.amazonaws.com/1466715971459-545595487b2d-9a33b32608e5/Tracker-Changed_Broken_proofs-545595487b2d.png) [(after)](http://keybase-app-visdiff.s3.amazonaws.com/1466715971459-545595487b2d-9a33b32608e5/Tracker-Changed_Broken_proofs-9a33b32608e5.png)  
   Tracker-Changed_Broken_proofs comparison

File: shared/tracker/dumb.native.js (hunk 6 above)
Comment (awendland) [inline shared/tracker/dumb.native.js]: I'll look into combining them as more of the mobile profile gets complete
